pkg/pilot/genericpilot/util.go

```diff
@@ -0,0 +1,47 @@
+package genericpilot
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/jetstack/navigator/pkg/apis/navigator/v1alpha1"
+)
+
+// IsThisPilot will return true if 'pilot' corresponds to the Pilot resource
+// for this pilot.
+func (g *GenericPilot) IsThisPilot(pilot *v1alpha1.Pilot) bool {
+	return pilot.Name == g.Options.PilotName && pilot.Namespace == g.Options.PilotNamespace
```

Comment:
❓ Should we check for same owner here too?

Reply:
The Pilot is not aware of the owner here, as we don't pass 'owning' information through. Instead, we pass through a pilot 'name and namespace' as args, and have the Pilots perform this comparison to 'find itself'.

After finding itself, it *then* checks the `ownerReferences` field to find out who the owning resource is.